client/go/outline/config/module.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2024 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"context"
+	"errors"
+	"net"
+
+	"github.com/Jigsaw-Code/outline-sdk/transport"
+)
+
+// TransportPair provides a StreamDialer and PacketListener, to use as the transport in a Tun2Socks VPN.
+type TransportPair struct {
```

Comment:
Do we really need this type? Why not just returning `(*Dialer[transport.StreamConn], *PacketListener, error)` in all functions?